src/app/components/ScrollOffset.tsx

```diff
@@ -0,0 +1,95 @@
+import { MouseEvent } from '@openstax/types/lib.dom';
+import React from 'react';
+import { createGlobalStyle, css } from 'styled-components/macro';
+import { isHtmlElement } from '../guards';
+import theme from '../theme';
+import { assertDocument, assertWindow } from '../utils';
+
+interface ScrollOffsetProps {
+  desktopOffset: number;
+  mobileOffset: number;
+}
+
+// tslint:disable-next-line:variable-name
+const GlobalStyle = createGlobalStyle<ScrollOffsetProps>`
+  body {
+    scroll-padding: ${(props) => props.desktopOffset * 10}px 0 0 0;
+    ${theme.breakpoints.mobile(css`
+      scroll-padding: ${(props: ScrollOffsetProps) => props.mobileOffset * 10}px 0 0 0;
+    `)}
+  }
+`;
+
+/*
+ * everything from here down is only necessary because some browsers (**ahem** safari 11, edge)
+ * do not support scroll-padding. safari *claims* to support scroll-padding, but doesn't outside
+ * the context of a scroll snap position
+ *
+ * https://bugs.webkit.org/show_bug.cgi?id=179379
+ */
+
+export default class ScrollOffset extends React.Component<ScrollOffsetProps> {
+
+  public getOffset = (window: Window) => -(window.matchMedia(theme.breakpoints.mobileQuery).matches
+    ? this.props.mobileOffset * 10
+    : this.props.desktopOffset * 10
```

Comment:
this won't work if the browser zoom is not default, not sure if there is a way to compensate for that

Reply:
or maybe it'll work for zoom but not if default font size has been changed 🤔